File: app/src/main/java/io/goooler/demoapp/util/ToastUtil.java
package io.goooler.demoapp.util;

import android.os.Looper;
import android.widget.Toast;

import androidx.annotation.StringRes;

import io.goooler.demoapp.base.BaseApplication;


/**
 * Toast 简单封装
 */

public class ToastUtil {

    /**
     * 可在子线程使用的 toast
     *
     * @param text string 文本
     */
    public static void showToast(String text) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            Toast.makeText(BaseApplication.getContext(), text, Toast.LENGTH_SHORT).show();
        } else {
            Looper.prepare();
            Toast.makeText(BaseApplication.getContext(), text, Toast.LENGTH_SHORT).show();
            Looper.loop();
        }
    }

    /**
     * 默认长度 Toast.LENGTH_SHORT
     *
     * @param stringId 文本资源 id
     */
    public static void showToast(@StringRes int stringId) {
        showToast(ResUtil.getString(stringId));
    }
}

File: app/src/main/java/io/goooler/demoapp/util/ToastUtil.kt
package io.goooler.demoapp.util

import android.content.Context
import android.os.Looper
import android.widget.Toast
import androidx.annotation.AnyThread
import androidx.annotation.MainThread


/**
 * Toast 简单封装
 */
object ToastUtil {

    /**
     * 可在子线程使用的 toast
     *
     * @param text string 文本
     */
    @AnyThread
    fun showToastInAnyThread(context: Context, text: String) {
        if (isMainThread()) {
            showToastInMainThread(context, text)
        } else {
            Looper.prepare()
            showToastInMainThread(context, text)
            Looper.loop()
        }
    }

    /**
     * 只在主线程调用
     */
    @MainThread
    fun showToastInMainThread(context: Context, text: String) {
        Toast.makeText(context, text, Toast.LENGTH_SHORT).show()
    }

    private fun isMainThread(): Boolean {
        return Looper.getMainLooper().thread === Thread.currentThread()
    }
}
